### PR TITLE
Support parsing `CREATE DIRECTORY` in openGauss

### PR DESCRIPTION
**Pull request by everly-gif**
Ref #17736 .

Changes proposed in this pull request:
- Adds `CREATE DIRECTORY` grammar
- Support parsing `CREATE DIRECTORY` in openGauss
- Adds Tests

✅ Builds locally

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/BaseRule.g4 [hunk 1]
```diff
@@ -1276,6 +1276,18 @@ aliasClause
     | colId
     ;
 
+directoryname
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/BaseRule.g4:1279]: Do you think `directoryName` is better?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/BaseRule.g4 [hunk 2]
```diff
@@ -1276,6 +1276,18 @@ aliasClause
...
+    : name
+    ;
+
+pathString
+    : stringLiterals
+    ;
+
+stringLiterals
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/BaseRule.g4:1287]: Can we remove this rule? And then modify `pathString` to `STRING_`.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/BaseRule.g4 (hunk 1 above)
Comment (everly-gif) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/BaseRule.g4]: agreed, we need to maintian consistency. made the changes

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/BaseRule.g4 (hunk 2 above)
Comment (everly-gif) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/BaseRule.g4]: sure.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**